### PR TITLE
 [ADD] [12.0] pos_restaurant_multi_company

### PR DESCRIPTION
**Pull request by legalsylvain**
trivial module

* add ``company_id`` field on ``pos_restaurant`` models (``restaurant.floor`` and ``restaurant.table`` (related) and ``restaurant.printer``)

* add classical rule

CC : @OCA/pos-maintainers 

Ref GRAP : apps/deck/#/board/144/card/1634 CC @quentinDupont

### CODE REVIEW

File: pos_restaurant_multi_company/models/restaurant_printer.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Copyright (C) 2021 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class RestaurantPrinter(models.Model):
+    _inherit = "restaurant.floor"
```
Comment (ivantodorovich) [inline pos_restaurant_multi_company/models/restaurant_printer.py]:
```suggestion
    _inherit = "restaurant.printer"
```

File: pos_restaurant_multi_company/models/restaurant_floor.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+class RestaurantFloor(models.Model):
+    _inherit = "restaurant.floor"
+
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        index=True,
+        default=lambda self:
+            self.env["res.company"]._company_default_get("restaurant.floor"),
```
Comment (ivantodorovich) [inline pos_restaurant_multi_company/models/restaurant_floor.py]: I would go for `default=lambda self: self.env.user.company_id`, here and in the other cases

To be honest I didn't know about `_company_default_get` but I see it's already deprecated in `14.0`: https://github.com/odoo/odoo/commit/a5b6f31cf28e5381e1c85f66730bcdb55998e643

I would also add a constraint to check, if `pos_config_id` is set, that they belong to the same company

Maybe even an onchange to automatically set the company from the pos.config, when set

File: pos_restaurant_multi_company/models/restaurant_printer.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        index=True,
+        default=lambda self:
+            self.env["res.company"]._company_default_get("restaurant.printer"),
+    )
```
Comment (ivantodorovich) [inline pos_restaurant_multi_company/models/restaurant_printer.py:16]: I think the `ir.rule` and views for the `restaurant.printer` are missing

**Review by ivantodorovich — CHANGES_REQUESTED** [review]
:+1:

File: pos_restaurant_multi_company/models/restaurant_printer.py (hunk 3 above)
Comment (legalsylvain) [reply pos_restaurant_multi_company/models/restaurant_printer.py:16]: you're right ! added.

File: pos_restaurant_multi_company/models/restaurant_floor.py (hunk 2 above)
Comment (legalsylvain) [reply pos_restaurant_multi_company/models/restaurant_floor.py]: To be honest, I don't know. I copied the code from other module of the same repo. the standards seemed to be ``_company_default_get``, but I think it's the same. So I simplified the code, and make it compatible with V14.

thanks !

Comment (legalsylvain) [reply pos_restaurant_multi_company/models/restaurant_floor.py]: constraint and onchange added.

File: pos_restaurant_multi_company/models/restaurant_printer.py (hunk 1 above)
Comment (legalsylvain) [reply pos_restaurant_multi_company/models/restaurant_printer.py]: done. thanks !

**Review by ivantodorovich — APPROVED** [review]
LGTM :+1: :tada:  Thanks !!

**Review by quentinDupont — APPROVED** [review]
Use in prod since more than one year, works like a charm :heavy_check_mark:

**Pull request merged**